service/controller/key/machine_deployment.go

```diff
@@ -33,10 +264,29 @@ func MachineDeploymentScalingMin(cr infrastructurev1alpha2.AWSMachineDeployment)
 	return cr.Spec.NodePool.Scaling.Min
 }
 
+func MachineDeploymentSpotInstancePools(overrides []template.LaunchTemplateOverride) int {
```

Comment:
Can we have some useful godoc here why this is a thing?

Reply:
Not sure what would be useful godoc here. This mainly exists to ensure that the resulting value is >= 1 and <= 20. These are AWS limitations and we want to prevent weird error behavior here.

Reply:
Sounds good. 😄